crates/flox/src/commands/environment.rs

```diff
@@ -52,11 +52,51 @@ impl Edit {
 
         let mut environment =
             resolve_environment(&flox, self.environment.as_deref(), "edit").await?;
-        let mut temporary_environment = environment.make_temporary()?;
-
         let nix = flox.nix(Default::default());
 
-        if let Some(file) = self.file {
+        match self.provided_manifest_contents()? {
+            // If provided with the contents of a manifest file, either via a path to a file or via
+            // contents piped to stdin, use those contents to try building the environment.
+            Some(new_manifest) => {
+                environment.edit(&nix, &flox.system, new_manifest).await?;
+                Ok(())
+            },
+            // If not provided with new manifest contents, let the user edit the file directly
+            // via $EDITOR (as long as `flox edit` was invoked interactively).
+            None => {
+                let editor = std::env::var("EDITOR").context("$EDITOR not set")?;
+                if !Dialog::can_prompt() {
```

Comment:
I'm not sure we should check this here - EDITOR may not be interactive (e.g. sometimes we use cat to print the manifest)

Reply:
Ah, I see `add-hello` where `EDITOR` is non-interactive.

There's actually two problems here:
1. How do you know whether you can prompt the user to continue editing?
2. How do you know whether `EDITOR` is interactive?

(1) is solved by checking that `stdin`/`stdout` are `tty`s, which is what `Dialog::can_prompt` is doing here.
I'm not sure if there's a solution for 2 that doesn't involve opening `EDITOR`. Once you open `EDITOR` I'm not sure if there's a way to know whether it's interactive or if you're now just stuck with an open interactive editor that will hang forever.

Reply:
Sorry I'm coming in with little context, but from what I understand you're trying to reproduce the logic found in the old `floxEdit()` function found in `flox-bash/lib/commands/environment.sh`. If you look at [lines 1003-1012](https://github.com/flox/flox/blob/main/flox-bash/lib/commands/environment.sh#L1003-L1012) you'll see that interactivity is only relevant when dealing with a failed EDITOR invocation; prior to that point you just invoke whatever the user has provided.

I'll also note that we haven't yet ported the [bestAvailableEditor](https://github.com/flox/flox/blob/main/flox-bash/lib/utils.sh#L179-L181) functionality from bash either. It would be great to fold that into this effort as well.

Reply:
The con of that is that EDITOR can hang forever. The pro is that it allows editing the manifest with a script. If we add a way to print the manifest contents, we could just do `flox print | script | flox edit` but maybe until then we can just invoke EDITOR as invoking `flox edit` noninteractively is likely uncommon?

Reply:
I think the script case is handled pretty well by the `stdin` case. I don't really see using `EDITOR` non-interactively as something that a user is likely to want to do.

Reply:
How would you get the current contents of the manifest for the script case?

Reply:
> How would you get the current contents of the manifest for the script case?

I think it would be pretty trivial to add a command that spits out the current manifest e.g. the `flox print` you have above.

> I guess we could add an additional check to only attempt to run flox edit interactively when the user has provided one of EDITOR or VISUAL in the environment

Maybe I'm misunderstanding but it sounds like this is exactly what the code above is doing (minus the check for `VISUAL`).

Reply:
I'm going to move the interactivity check for now and leave a `TODO` item where the current check is. I created #246 so that we can come back to this and put the original check back in. I consider it a bug that a user can call `flox edit` interactively in a non-interactive context.